viadot/sources/customer_gauge.py

```diff
@@ -0,0 +1,250 @@
+        end_date: datetime = None,
+    ) -> Dict[str, Any]:
+        """
+        Gets JSON with nested structure that contains data and cursor parameter value using GET request method.
+
+        Args:
+            url (str, optional): Endpoint URL. Defaults to None.
+            cursor (int, optional): Cursor value to navigate to the page. Defaults to None.
+            pagesize (int, optional): Number of responses (records) returned per page, max value = 1000. Defaults to 1000. Defaults to 1000.
+            date_field (Literal["date_creation", "date_order", "date_sent", "date_survey_response"], optional): Specifies the date type which filter date range. Defaults to None.
+            start_date (datetime, optional): Defines the period start date in yyyy-mm-dd format. Defaults to None.
+            end_date (datetime, optional): Defines the period end date in yyyy-mm-dd format.. Defaults to None.
+
+        Raises:
+            ValueError: If at least one date argument were provided and the rest is missing. Needed all 3 or skip them.
+            APIError: If no response from API call.
+
+        Returns:
+            Dict[str, Any]: JSON with data and cursor parameter value.
+        """
+        if url is None:
+            url = self.url
+
+        params = {
+            "per_page": pagesize,
+            "with[]": ["drivers", "tags", "questions", "properties"],
+            "cursor": cursor,
+        }
+
+        if any([date_field, start_date, end_date]):
+            if all([date_field, start_date, end_date]):
+                params["period[field]"] = (date_field,)
+                params["period[start]"] = (start_date,)
+                params["period[end]"] = end_date
+            else:
+                raise ValueError(
+                    "Missing date arguments: 'date_field', 'start_date', 'end_date'. Provide all 3 arguments or skip all of them."
+                )
+
+        header = {"Authorization": f"Bearer {self.get_token()}"}
+        api_response = handle_api_response(url=url, headers=header, params=params)
+        response = api_response.json()
+
+        if response is None:
+            raise APIError("No response.")
+        return response
+
+    def get_cursor(self, json_response: Dict[str, Any] = None) -> int:
+        """
+        Returns cursor value that is needed to navigate to the next page in the next API call for specific pagesize.
+
+        Args:
+            json_response (Dict[str, Any], optional): Dictionary with nested structure that contains data and cursor parameter value. Defaults to None.
+
+        Raises:
+            ValueError: If cursor value not found.
+
+        Returns:
+            int: Cursor value.
+        """
+
+        try:
+            cur = json_response["cursor"]["next"]
+        except:
+            raise ValueError(
+                "Provided argument doesn't contain 'cursor' value. Pass json returned from the endpoint."
+            )
+
+        return cur
+
```

Comment:
Between `return` and `def new_function` should be 2 empty lines. Please correct in the entire file.

Reply:
In VSCode after saving it removed extra line I'm adding 😖